packages/big-design/src/components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,103 @@
+import { ArrowDropDownIcon, ChevronLeftIcon, ChevronRightIcon } from '@bigcommerce/big-design-icons';
+import React, { useEffect, useState } from 'react';
+
+import { MarginProps } from '../../mixins';
+import { Dropdown } from '../Dropdown';
+import { Flex } from '../Flex';
+
+import { StyledButton } from './styled';
+
+export interface PaginationProps extends MarginProps {
+  currentPerPage: number;
+  currentPage: number;
+  numOfItems: number;
+  perPageOptions: number[];
+  onPageUpdate(page: number): void;
+  onRangeUpdate(range: number): void;
+}
+
+export const Pagination: React.FunctionComponent<PaginationProps> = props => {
+  const { currentPerPage, currentPage, numOfItems, perPageOptions, onPageUpdate, onRangeUpdate } = props;
+
+  const [maxPages, setMaxPages] = useState(Math.ceil(numOfItems / currentPerPage));
+  const [itemRange, setItemRange] = useState([0, 0]);
+
+  useEffect(() => {
+    const first = currentPerPage * (currentPage - 1) + 1;
+    let second = currentPerPage * currentPage;
+    if (second > numOfItems) {
+      second = numOfItems;
+    }
+    const range = [first, second];
+    setItemRange(range);
+  }, [currentPage, currentPerPage, numOfItems]);
+
+  const handlePageIncrease = () => {
+    let nextPage = currentPage + 1;
+    if (nextPage > maxPages) {
+      nextPage = 1;
+    }
+    onPageUpdate(nextPage);
+  };
+
+  const handlePageDecrease = () => {
+    let nextPage = currentPage - 1;
+    if (nextPage < 1) {
+      nextPage = maxPages;
+    }
+    onPageUpdate(nextPage);
+  };
+
+  const handleRangeChange = (range: number) => {
+    setMaxPages(Math.ceil(numOfItems / range));
+    onRangeUpdate(range);
+  };
+
+  const showRanges = () => {
+    if (itemRange[0] !== itemRange[1]) {
+      return itemRange[0] + ' - ' + itemRange[1] + ' of ' + numOfItems;
+    } else {
+      return itemRange[0] + ' of ' + numOfItems;
+    }
+  };
+
+  return (
+    <Flex flexDirection="row" justifyContent="space-between" role="navigation" aria-label="pagination">
+      <Dropdown
+        onItemClick={handleRangeChange}
+        trigger={
+          <StyledButton>
+            {showRanges()}
+            <ArrowDropDownIcon color="secondary70" size="xxLarge" />
+          </StyledButton>
+        }
+      >
+        {perPageOptions.map(range => {
+          return <Dropdown.Item value={range}>{range} per page</Dropdown.Item>;
+        })}
+      </Dropdown>
+      {currentPage > 1 ? (
```

Comment:
I would wrap these icons in a button, specifically one you extend from above comment, so you don't have to manually deal with the accessibility stuff, plus you can determine `disabled` through the button prop rather than a conditional statement.